src/clj/game/core/memory.clj

```diff
@@ -137,14 +139,29 @@
        (swap! state update-in [:runner :memory] merge new-mu))
      changed?)))
 
+(defn some-mu-effect?
+  [state card]
+  (let [ab (first (filter #(= (:type %) :used-mu) (:static-abilities (card-def card))))
+        abreq (:req ab)
+        abval (:value ab)]
+    (if (and ab (or (nil? abreq) (abreq state :runner nil card nil)))
+      (abval state :runner nil card nil)
+      0)))
+
+(defn expected-mu
+  [state card]
+  (when (program? card)
+    (+ (:memoryunits card) (some-mu-effect? state card))))
```

Comment:
This returns `nil` if it's not a program, so you either need to guard against nils everywhere it's used or maybe default to 0 if not a program. If guarding, use `keep` instead of map: `(reduce + 0 (keep expected-mu some-coll))`.